ironfish/src/network/peers/peerAddressManager.ts

```diff
@@ -0,0 +1,21 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { HostsStore } from '../../fileStores'
+import { PeerAddress } from './peerAddress'
+
+export class PeerAddressManager {
+  hostsStore: HostsStore
+
+  constructor(hostsStore: HostsStore) {
```

Comment:
ok, if this thing isn't the authority on its persistence, who is? The SDK, the node?

Reply:
Yep, there are modifications to both the node and SDK.